piwik-sdk/src/main/java/org/piwik/sdk/tools/DeviceHelper.java

```diff
@@ -44,7 +50,19 @@ public String getUserLanguage() {
      * @return well formatted user agent
      */
     public String getUserAgent() {
-        return System.getProperty("http.agent");
+        String httpAgent = mPropertySource.getHttpAgent();
+        if (httpAgent == null || httpAgent.startsWith("Apache-HttpClient/UNAVAILABLE (java")) {
+            String dalvik = mPropertySource.getJVMVersion();
+            if (dalvik == null) dalvik = "0.0.0";
+            String android = mBuildInfo.getRelease();
+            String model = mBuildInfo.getModel();
+            String build = mBuildInfo.getBuildId();
+            httpAgent = String.format(Locale.US,
+                    "Dalvik/%s (Linux; U; Android %s; %s Build/%s)",
```

Comment:
@sgiehl Thoughts on the format? Seems to be the most common Android http agent format.